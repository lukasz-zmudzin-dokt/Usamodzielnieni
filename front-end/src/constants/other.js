const DEFAULT_INPUT = "-- Wybierz --";

export { DEFAULT_INPUT };
