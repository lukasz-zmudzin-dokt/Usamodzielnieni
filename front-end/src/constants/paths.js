const paths = {
    DASHBOARD: "/",
    CV_CREATOR: "/cvEditor",
    CV_EDITOR: "/cvEditor/:id",
    REGISTER: "/newAccount/",
    REGISTER_ADMIN: "/newAccount/:role(staff)",
    FOOTER: "/footer",
    LOGIN: "/login",
    USER: "/user",
    OFFER_FORM: "/offerForm/:id?",
    JOB_OFFERS: "/jobOffers",
    CV_APPROVAL: "/cvApproval",
    MY_OFFERS: "/myOffers",
    CONTACT_PAGE: "/contact",
    BLOG_POST: "/blog/blogpost/:id",
    BLOG_PAGE: "/blog",
    BLOG_FORM: "/blog/newPost",
    BLOG_EDIT: "/blog/newPost/:id",
    JOB_OFFER_DETAILS: "/jobOffers/:id",
    MY_CVS: "/myCVs",
    USER_APPROVAL: "/userApproval",
    OFFER_APPROVAL: "/offerApproval"
};
export {paths};