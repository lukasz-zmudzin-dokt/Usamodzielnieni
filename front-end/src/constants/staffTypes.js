export const staffTypes = {
  VERIFICATION: "staff_verification",
  CV: "staff_cv",
  JOBS: "staff_jobs",
  BLOG_CREATOR: "staff_blog_creator",
  BLOG_MODERATOR: "staff_blog_moderator",
};
