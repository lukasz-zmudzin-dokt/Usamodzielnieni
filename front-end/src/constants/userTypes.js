export const userTypes = {
  STANDARD: "standard",
  STAFF: "staff",
  EMPLOYER: "employer",
};
