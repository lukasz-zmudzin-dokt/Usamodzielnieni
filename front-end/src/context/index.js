export { UserContext, UserProvider } from "context/UserContext";
export { AlertContext, AlertProvider } from "context/AlertContext";
export { ChatContext, ChatProvider } from "context/ChatContext";
export {
  NotificationsContext,
  NotificationsProvider,
} from "context/NotificationsContext";
