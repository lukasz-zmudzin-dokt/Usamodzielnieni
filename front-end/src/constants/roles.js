import { staffTypes } from "./staffTypes";

export const adminGroup = [
  {
    name: staffTypes.VERIFICATION,
    placeholder: "Weryfikacja użytkowników",
  },
  {
    name: staffTypes.CV,
    placeholder: "Weryfikacja CV",
  },
  {
    name: staffTypes.JOBS,
    placeholder: "Weryfikacja ofert pracy",
  },
  {
    name: staffTypes.BLOG_CREATOR,
    placeholder: "Kreator postów na blogu",
  },
  {
    name: staffTypes.BLOG_MODERATOR,
    placeholder: "Moderator bloga / designer",
  },
  {
    name: staffTypes.GUEST,
    placeholder: "Konto gościa",
  },
  {
    name: staffTypes.CHAT,
    placeholder: "Dostęp do czatu",
  },
];

export const commonGroup = [
  {
    name: "Podopiecznym",
  },
  {
    name: "Pracodawcą",
  },
];
