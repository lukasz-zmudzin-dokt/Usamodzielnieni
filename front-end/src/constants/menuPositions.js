import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
