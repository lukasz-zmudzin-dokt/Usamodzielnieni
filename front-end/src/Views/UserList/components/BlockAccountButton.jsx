import React, { useState, useContext } from "react";
import { Button } from "react-bootstrap";
import { DeletionModal } from "components";
import proxy from "config/api";
import { AlertContext, UserContext } from "context";
import { userStatuses } from "constants/userStatuses";
import {staffTypes} from "constants/staffTypes";

const blockAccount = async (token, user) => {
  let url = `${proxy.account}admin/user_block/${user.id}/`;
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };

  const response = await fetch(url, { method: "POST", headers });

  if (response.status !== 200) {
    throw response.status;
  }
};

const BlockAccountButton = ({ user, setUser, ...rest }) => {
  const [showModal, setShowModal] = useState(false);
  const [disabled, setDisabled] = useState(false);

  const { token, data } = useContext(UserContext);
  const alertContext = useContext(AlertContext);

  const onButtonClick = () => setShowModal(true);
  const onSuccess = async () => {
    setDisabled(true);
    try {
      await blockAccount(token, user);
    } catch (e) {
      console.log(e);
      alertContext.showAlert("Wystąpił błąd podczas blokowania konta.");
      setDisabled(false);
      return;
    }
    alertContext.showAlert("Pomyślnie zablokowano konto.", "success");
    setDisabled(false);
    setUser({
      ...user,
      status: userStatuses.BLOCKED,
    });
  };

  return (
      data.group_type.includes(staffTypes.BLOG_MODERATOR) ||
      data.group_type.includes(staffTypes.VERIFICATION)
  ) && (
    <>
      <Button
        variant="secondary"
        onClick={onButtonClick}
        disabled={disabled}
        {...rest}
      >
        {disabled ? "Ładowanie..." : "Zablokuj konto"}
      </Button>
      <DeletionModal
        show={showModal}
        setShow={setShowModal}
        delConfirmed={onSuccess}
        question="Czy na pewno chcesz zablokować to konto?"
        confirmLabel="Zablokuj"
        cancelLabel="Anuluj"
      />
    </>
  );
};

export default BlockAccountButton;
