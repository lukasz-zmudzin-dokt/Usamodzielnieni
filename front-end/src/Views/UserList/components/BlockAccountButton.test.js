import React from "react";
import { render, fireEvent, wait, queries } from "@testing-library/react";
import BlockAccountButton from "./BlockAccountButton";
import { AlertContext, UserContext } from "context";
import {userTypes} from "constants/userTypes";
import {staffTypes} from "constants/staffTypes";

describe("BlockAccountButton", () => {
  let apiShouldFail;
  let user;
  let contextA;
  let setUser;
  let context;
  beforeAll(() => {
    global.fetch = jest.fn().mockImplementation((input, init) => {
      return new Promise((resolve, reject) => {
        if (apiShouldFail) {
          resolve({ status: 500 });
        } else {
          switch (init.method) {
            case "POST":
              resolve({ status: 200 });
              break;
            default:
              reject({});
              break;
          }
        }
      });
    });
  });
  beforeEach(() => {
    apiShouldFail = false;
    user = {
      token: "123",
      logout: jest.fn(),
    };
    contextA = {
      showAlert: jest.fn(),
    };
    setUser = jest.fn();
    context = {
      type: userTypes.STAFF,
      data: {
        group_type: [staffTypes.VERIFICATION, staffTypes.BLOG_MODERATOR]
      }
    }
  });

  it("should render without crashing", async () => {
    const { container } = render(
        <UserContext.Provider value={context}>
          <BlockAccountButton user={user} />
        </UserContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should open modal when button is clicked", async () => {
    const { getByText } = render(
        <UserContext.Provider value={context}>
          <BlockAccountButton user={user} />
        </UserContext.Provider>
    );

    fireEvent.click(getByText("Zablokuj konto", { exact: false }));

    expect(
      getByText("Czy na pewno chcesz zablokować to konto?")
    ).toBeInTheDocument();
  });

  it("should display success alert when confirmation button is clicked", async () => {
    const { getByText, getByRole, queryByText } = render(
      <AlertContext.Provider value={contextA}>
        <UserContext.Provider value={context}>
          <BlockAccountButton user={user} setUser={setUser}/>
        </UserContext.Provider>
      </AlertContext.Provider>
    );

    fireEvent.click(getByText("Zablokuj konto", { exact: false }));

    const modal = getByRole("dialog");
    fireEvent.click(queries.getByText(modal, "Zablokuj", { exact: false }));

    expect(getByText("Ładowanie...")).toBeInTheDocument();

    await wait(() =>
      expect(
        queryByText("Czy na pewno chcesz zablokować to konto?", {
          exact: false,
        })
      ).not.toBeInTheDocument()
    );

    expect(contextA.showAlert).toHaveBeenCalledWith(
      "Pomyślnie zablokowano konto.",
      "success"
    );
  });

  it("should display error alert when api fails", async () => {
    apiShouldFail = true;
    const { getByText, getByRole, queryByText } = render(
      <AlertContext.Provider value={contextA}>
        <UserContext.Provider value={context}>
          <BlockAccountButton user={user} setUser={setUser}/>
        </UserContext.Provider>
      </AlertContext.Provider>
    );

    fireEvent.click(getByText("Zablokuj konto", { exact: false }));

    const modal = getByRole("dialog");
    fireEvent.click(queries.getByText(modal, "Zablokuj", { exact: false }));

    expect(getByText("Ładowanie...")).toBeInTheDocument();
    await wait(() =>
      expect(
        queryByText("Czy na pewno chcesz zablokować to konto?", {
          exact: false,
        })
      ).not.toBeInTheDocument()
    );

    expect(contextA.showAlert).toHaveBeenCalledWith(
      "Wystąpił błąd podczas blokowania konta."
    );
    expect(getByText("Zablokuj konto")).toBeInTheDocument();
  });
});
