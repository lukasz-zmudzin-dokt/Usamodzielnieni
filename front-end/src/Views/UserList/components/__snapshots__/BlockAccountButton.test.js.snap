// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockAccountButton should render without crashing 1`] = `
<div>
  <button
    class="btn btn-secondary"
    type="button"
  >
    Zablokuj konto
  </button>
</div>
`;
